File: WatchWhat-iOS/Swift-App/Model/MovieDb/Movie.swift
//
//  MovieObject.swift
//  Swift-App
//
//  Created by Drew Miller on 11/7/22.
//

import Foundation
import SwiftUI

struct MovieResponse: Codable {
    let page: Int
    let results: [Movie]
    let totalPages: Int
    let totalResults: Int
}

struct Movie: TmdbImage, MovieDate, Hashable, Codable {
    let adult: Bool
    let backdropPath: String
    let genreIds: [Int]
    let id: Int
    let originalLanguage: String
    let originalTitle: String
    let overview: String
    let popularity: Float
    let posterPath: String
    let releaseDate: String?
    let title: String
    let video: Bool
    let voteAverage: Float
    let voteCount: Int
    
    var backdropUrl: URL? {
        return getImageURL(backdropPath)
    }
    
    var posterUrl: URL? {
        return getImageURL(posterPath)
    }
    
    var release: Date? {
        return formatDate(releaseDate)
    }
    
    var year: String {
        return getYear(releaseDate)
    }
};

File: WatchWhat.GraphQL/Shared/tmdb/dtos/movie/movie.ts
import { Company, Genre, SpokenLanguages } from "../.";

// This property is likely a collection group.
// belongs_to_collection: any | null,
export type Movie = {
  adult: boolean,
  backdrop_path?: string,
  budget: number,
  id: number,
  genres: Genre[],
  homepage: string,
  imdb_id: string,
  original_language: string,
  original_title: string,
  overview: string,
  popularity: number,
  poster_path: string,
  production_companies: Company[],
  production_countries: Company[],
  release_date?: Date,
  revenue: number,
  runtime: number,
  spoken_languages: SpokenLanguages[],
  status: string,
  tagline: string,
  title: string,
  video: boolean,
  vote_average: number,
  vote_count: number
};